imports/ui/unit-explorer/unit-explorer.jsx

```diff
@@ -39,55 +42,83 @@ class UnitExplorer extends Component {
     dispatch(push(`/unit/${id}`))
   }
 
+  findUnit = (searchText) => {
+    if (searchText === '') {
+      this.setState({searchEnabled: false})
+    } else {
+      this.setState({searchEnabled: true})
+      const matcher = new RegExp(searchText, 'i')
+      const searchResult = this.props.unitList
+        .filter(unit => !matcher || (unit.name && unit.name.match(matcher)))
+      this.setState({
+        searchResult: searchResult
+      })
+    }
+  }
+
+  searchOff = () => {
+    this.setState({searchEnabled: false})
+  }
+
   render () {
     const { isLoading, unitList, dispatch, currentUserId } = this.props
-
+    const { searchResult, searchEnabled } = this.state
     if (isLoading) return <Preloader />
     const activeUnits = unitList.filter(unitItem => unitItem.metaData && !unitItem.metaData.disabled)
     const disabledUnits = unitList.filter(unitItem => unitItem.metaData && unitItem.metaData.disabled)
-
     return (
       <div className='flex flex-column flex-grow full-height'>
-        <RootAppBar title='My Units' onIconClick={() => dispatch(setDrawerState(true))} shadowless />
+        <RootAppBar
+          title='My Units'
+          onIconClick={() => dispatch(setDrawerState(true))}
+          shadowless
+          findUnit={this.findUnit}
+          searchOff={this.searchOff}
+        />
         <UnverifiedWarning />
-        <div className='flex-grow flex flex-column overflow-hidden'>
-          <Tabs
-            className='no-shrink'
-            onChange={this.handleChange}
-            value={this.state.slideIndex}
-            inkBarStyle={{backgroundColor: 'white'}}
-          >
-            <Tab label='Active' value={0} />
-            <Tab label='Disabled' value={1} />
-          </Tabs>
-
-          <div className='flex-grow flex flex-column overflow-auto'>
-            <SwipeableViews
-              index={this.state.slideIndex}
-              onChangeIndex={this.handleChange}
+        { searchEnabled ? (
+          <SearchResult unitsFound={searchResult}
```

Comment:
Shouldn't FilteredUnitsList be used over this new component? they seem practically the same